### PR TITLE
Added DefaultLibraryNamespace() function

### PR DESCRIPTION
**Pull request by ewoutp**
Added function to DockerImage that returns a derived image that sets the Namespace to "library" when it was empty.

https://github.com/giantswarm/app-service/pull/679

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 .gobuild/
+generic-types-go
```
Comment (ewoutp) [inline .gitignore]: Needed for go 1.5

**Comment by ewoutp** [comment]
@zeisss RFR

File: image_test.go [hunk 2]
```diff
@@ -303,3 +303,27 @@ func TestVersionDefaulting(t *testing.T) {
 		t.Fatalf("Expected image version to be unchanged, got '%s'", latestImage.Version)
 	}
 }
+
+func TestNamespaceDefaulting(t *testing.T) {
+	img, _ := ParseDockerImage("redis")
```
Comment (zeisss) [inline image_test.go]: MustParseDockerImage

File: image_test.go [hunk 3]
```diff
@@ -303,3 +303,27 @@ func TestVersionDefaulting(t *testing.T) {
...
+
+	if img.Namespace != "" {
+		t.Fatalf("Expected parsed image namespace to empty, got '%s'", img.Namespace)
+	}
+
+	if libraryImage.Namespace != "library" {
+		t.Fatalf("Expected namespace to be 'library', got '%s'", libraryImage.Namespace)
+	}
+
+	img, _ = ParseDockerImage("mynamespace/redis")
```
Comment (zeisss) [inline image_test.go]: MustParseDockerImage

**Comment by zeisss** [comment]
I would use `MustParseDockerImage`, otherwise LGTM

**Comment by ewoutp** [comment]
Using MustParseDockerImage now for old (default version) and new (default namespace)

**Pull request merged**